### DISCUSSION TITLE
A component of the virtual machine is not accessible on the host

### DISCUSSION THREAD

**Discussion by macsmax** [vmware/packer-examples-for-vsphere#32] [Help] [answered]
Hello,

Thanks for the project. I am trying to build an ubuntu 20.04 image and I get the following error:
```Build 'vsphere-iso.linux-ubuntu-server' errored after 17 seconds 249 milliseconds: error creating vm: A component of the virtual machine is not accessible on the host.```
I tried to run packer in -debug mode but not much more info is given.
Packer version: 1.7.4
Vsphere version: 6.7 single ESXi host, no cluster

**Comment by tenthirtyam** [comment]
Hi! This project is designed to provide working examples for using Packer with a vSphere environment that includes a vCenter Server instance. It uses the objects, such as, datacenter, cluster, and content library.

Can you confirm your environment meets these requirements?

Thanks,
Ryan

**Comment by macsmax** [comment]
Hi Ryan,
I do have a vCenter Server with a single esx host, no cluster. I have changed the packer config as follow to adapt to use host rather than cluster:
```diff --git a/builds/linux/ubuntu-server-20-04-lts/ubuntu-server-20-04-lts.pkr.hcl b/builds/linux/ubuntu-server-20-04-lts/ubuntu-server-20-04-lts.pkr.hcl
index 9f9472e..8765188 100644
--- a/builds/linux/ubuntu-server-20-04-lts/ubuntu-server-20-04-lts.pkr.hcl
+++ b/builds/linux/ubuntu-server-20-04-lts/ubuntu-server-20-04-lts.pkr.hcl
@@ -351,7 +351,8 @@ source "vsphere-iso" "linux-ubuntu-server" {

   // vSphere Settings
   datacenter = var.vsphere_datacenter
-  cluster    = var.vsphere_cluster
+  // cluster    = var.vsphere_cluster
+  host    = var.vsphere_cluster
   datastore  = var.vsphere_datastore
   folder     = var.vsphere_folder
```

**Reply by tenthirtyam** [reply]
Does the `folder` specified for the `vsphere_folder` variable already exist?

Ryan

**Reply by macsmax** [reply]
Hey Ryan, 
yes the folder is there.
I created a cluster on my vcsa, moved the host in there. I should now have all needed: datacenter, cluster, and content library but unfortunately I still see the same error:
```
$ packer build -debug -force --only vsphere-iso.linux-ubuntu-server -var-file="../../vsphere.pkrvars.hcl" -var-file="../../build.pkrvars.hcl" -var-file="../../ansible.pkrvars.hcl" -var-file="../../common.pkrvars.hcl" .
Debug mode enabled. Builds will not be parallelized.
vsphere-iso.linux-ubuntu-server: output will be in this color.

==> vsphere-iso.linux-ubuntu-server: Pausing after run of step 'StepConnect'. Press enter to continue.
==> vsphere-iso.linux-ubuntu-server: Pausing after run of step 'StepDownload'. Press enter to continue.
==> vsphere-iso.linux-ubuntu-server: Pausing after run of step 'StepCreateCD'. Press enter to continue.
==> vsphere-iso.linux-ubuntu-server: Pausing after run of step 'StepRemoteUpload'. Press enter to continue.
==> vsphere-iso.linux-ubuntu-server: Creating VM...
==> vsphere-iso.linux-ubuntu-server: Pausing before cleanup of step 'StepRemoteUpload'. Press enter to continue.
==> vsphere-iso.linux-ubuntu-server: Pausing before cleanup of step 'StepCreateCD'. Press enter to continue.
==> vsphere-iso.linux-ubuntu-server: Pausing before cleanup of step 'StepDownload'. Press enter to continue.
==> vsphere-iso.linux-ubuntu-server: Pausing before cleanup of step 'StepConnect'. Press enter to continue.
Build 'vsphere-iso.linux-ubuntu-server' errored after 5 seconds 25 milliseconds: error creating vm: A component of the virtual machine is not accessible on the host.

==> Wait completed after 5 seconds 25 milliseconds

==> Some builds didn't complete successfully and had errors:
--> vsphere-iso.linux-ubuntu-server: error creating vm: A component of the virtual machine is not accessible on the host.

==> Builds finished but no artifacts were created.
```
Any tips?

Thanks for your help.
Max

**Comment by tenthirtyam** [comment]
Since you are running on vSphere 6.7 ([the repo is designed for 7.0 U2+](https://github.com/rainpole/packer-vsphere/#Requirements)) did you downlevel the hardware version from 18 down to 14 (6.7 GA) or 16 (6.7 U2) in `/builds/common.pkvars.hcl`?

Ryan

**Comment by macsmax** [comment]
I tried both 14 and 16, same error. I'll dig a bit deeper and revert back.

**Comment by tenthirtyam** [comment]
Any luck?

**Comment by tenthirtyam** [comment, marked as answer]
I was able to recreate the issue by bumping the to a version that does not exist in vSphere 7.0 U2.

![image](https://user-images.githubusercontent.com/7771363/133343253-ee35e8c4-6858-444d-9456-6ffc95c0f80b.png)

This resulted in the exact same error.

![image](https://user-images.githubusercontent.com/7771363/133343168-7a24a76b-a2cd-44cb-84e6-72e6d06e7719.png)

Refer to the following for supported version numbers: https://docs.vmware.com/en/VMware-vSphere/7.0/com.vmware.vsphere.vm_admin.doc/GUID-789C3913-1053-4850-A0F0-E29C3D32B6DA.html

Ryan
